### PR TITLE
Update how multipliers are stored

### PR DESCRIPTION
**Pull request by 0xLucca**
Dont merge yet

### CODE REVIEW

File: contracts/lib.rs [hunk 1]
```diff
@@ -123,28 +214,31 @@ mod open_payroll {
...
         #[ink(message)]
         pub fn add_or_update_beneficiary(
             &mut self,
             account_id: AccountId,
-            multipliers: Vec<Multiplier>,
+            multipliers: Vec<(MultiplierId,Multiplier)>,
         ) -> Result<(), Error> {
             self.ensure_owner()?;
 
             // Check that the multipliers are valid and have the same length as the base_multipliers
```
Comment (0xLucca) [inline contracts/lib.rs:236]: Add check to ensure that multiplier_id exists in base multipliers and also ensure chat multiplier_ids are not repeated in the vec

File: contracts/errors.rs [hunk 2]
```diff
@@ -23,4 +23,14 @@ pub enum Error {
...
     // The amount of multipliers per Beneficiary is not equal to the amount of periods
     InvalidMultipliersLength,
+    // The multiplier id does not exist
+    MultiplierNotFound,
+    // The multiplier is already deactivated
+    MultiplierAlreadyDeactivated,
+    // The multiplier is not deactivated
+    MultiplierNotDeactivated,
+    // The multiplier is not deactivated
+    MultiplierStillInUse,
```
Comment (0xLucca) [inline contracts/errors.rs:33]: MultiplierNotDeactivated and MultiplierStillInUse mean the same thing. Join them

File: contracts/lib.rs [hunk 3]
```diff
@@ -123,28 +216,67 @@ mod open_payroll {
...
+                    return Err(Error::MultiplierAlreadyDeactivated);
+                }
+            }
+            Ok(())
+        }
+
+        //TODO: Maybe this function could be generic over the type of the vec and the error type
+        // Can be used to check unique multipliers and also unique beneficiaries
+        fn check_no_duplicate_multipliers(
+            &self,
```
Comment (0xLucca) [inline contracts/lib.rs:251]: This function should be pure? Not depend on self. Because just uses the parameter

File: contracts/lib.rs [hunk 4]
```diff
@@ -50,39 +68,69 @@ mod open_payroll {
         initial_block: u32,
         /// The block number when the contract was paused
         paused_block_at: Option<u32>,
+        /// The id of the next multiplier to be added
+        next_multiplier_id: MultiplierId,
         /// The multipliers to apply to the base payment
-        base_multipliers: Vec<String>,
+        base_multipliers: Mapping<MultiplierId, BaseMultiplier>,
```
Comment (0xLucca) [inline contracts/lib.rs:74]: Review if a Mapping its ok or its better to be a Btreemap

File: contracts/lib.rs (hunk 1 above)
Comment (rtomas) [reply contracts/lib.rs]: this is already solved ?

File: contracts/lib.rs [hunk 5]
```diff
@@ -405,8 +567,135 @@ mod open_payroll {
...
+                }
+            }
+
+            total
+        }
+
+        /// get unclaimed balance per beneficiary
+        /// read-only
+        #[ink(message)]
+        pub fn get_unclaimed_balance(&self, account_id: AccountId) -> Balance {
```
Comment (0xLucca) [inline contracts/lib.rs:672]: We should remove this function, it does the same thing as get_amount_to_claim

**Pull request merged**